tests/unit/test_finder.py

```diff
@@ -701,3 +701,44 @@ def test_link_package_versions_substring_fails(self):
         link = Link('http:/yo/pytest_xdist-1.0-py2.py3-none-any.whl')
         result = self.finder._link_package_versions(link, self.search_name)
         assert result == [], result
+
+
+def test_get_indexes_location():
+    """Check that the canonical name is on all indexes"""
+    finder = PackageFinder(
+        [], ['file://index1/', 'file://index2'], session=PipSession())
+    locations = finder._get_indexes_locations(
```

Comment:
just noting that I saw this comment.   I don't have a solid opinion on this.   maybe another PR to do this kind of work if you feel strongly about it, where the debate can be just about that.